### DISCUSSION TITLE
LDM Notes for July 22nd, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#8310] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-07-22.md

## Agenda

- Extensions
- Ref structs implementing interfaces

**Comment by qrli** [comment]
> if the interface is updated, the implementor needs to go recompile with the new interface member and publish a new version.

What's the behavior if I use a library which has not been recompiled with new interface definition? Will I encounter error at compile time or at run time?

>  the language will allow [UnscopedRef] to appear on interface methods and properties. 

Is it a breaking change to add `[UnscopedRef]` to an interface member? My gut feeling is no, but I'm not sure.

**Reply by 333fred** [reply]
> What's the behavior if I use a library which has not been recompiled with new interface definition? Will I encounter error at compile time or at run time?

Runtime.

> Is it a breaking change to add [UnscopedRef] to an interface member?

That's no different than adding unscopedref in any other location after the fact.
